mypy/main.py

```diff
@@ -528,6 +514,24 @@ def add_invertible_flag(flag: str,
         print("Warning: --no-fast-parser no longer has any effect.  The fast parser "
               "is now mypy's default and only parser.")
 
+    # Infer Python version and/or executable if one is not given
+    if special_opts.python_executable is not None and special_opts.python_version is not None:
+        py_exe_ver = _python_version_from_executable(special_opts.python_executable)
+        if py_exe_ver != special_opts.python_version:
+            print('Error: Python version and executable are mismatched.')
+            sys.exit(2)
+        else:
+            options.python_version = special_opts.python_version
+            options.python_executable = special_opts.python_executable
+    elif special_opts.python_executable is None and special_opts.python_version is not None:
+        py_exe = _python_executable_from_version(special_opts.python_version)
+        if py_exe is not None:
+            options.python_executable = py_exe
```

Comment:
`_python_executable_from_version` should never return `None`. Either you somehow found an executable that matched the version, or you couldn't find it and need to ask the user to specify it.

Reply:
Hm, I disagree. If I require an executable I can't find this could break a lot of people's setups if they are using `--python-version`.

Reply:
Though I suppose the current state of just silently ignoring that is not ideal. Perhaps a warning is a good compromise.